curriculum/challenges/english/25-front-end-development/quiz-javascript-functional-programming/66edd4f31ff19bf5573bf64b.md

```diff
@@ -17,439 +17,472 @@ Answer all of the questions below correctly to pass the quiz.
+A function that always returns the same output for the same input
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is a characteristic of a pure function?
 
 #### --distractors--
 
-Placeholder distractor 1
+It has side effects
 
 ---
 
-Placeholder distractor 2
+It relies on global state
 
 ---
 
-Placeholder distractor 3
+It modifies its input parameters
 
 #### --answer--
 
-Placeholder answer
+It's output depends solely on its input arguments
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a side effect in functional programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+An unexpected error in the code
 
 ---
 
-Placeholder distractor 2
+A function that takes too long to execute
 
 ---
 
-Placeholder distractor 3
+A recursive function call
 
 #### --answer--
 
-Placeholder answer
+A change in program state or interaction with the outside world
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is an example of a side effect?
 
 #### --distractors--
 
-Placeholder distractor 1
+Returning a value from a function
 
 ---
 
-Placeholder distractor 2
+Creating a new object within a function
 
 ---
 
-Placeholder distractor 3
+Using a local variable
 
 #### --answer--
 
-Placeholder answer
+Writing to a file
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is currying in functional programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+A technique for cooking functions
 
 ---
 
-Placeholder distractor 2
+A method for optimizing recursive functions
 
 ---
 
-Placeholder distractor 3
+A way to combine multiple functions into one
 
 #### --answer--
 
-Placeholder answer
+The process of transforming a function with multiple arguments into a sequence of functions, each with a single argument
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In the context of pure functions, what does "referential transparency" mean?
 
 #### --distractors--
 
-Placeholder distractor 1
+The function's code is visible to other parts of the program
 
 ---
 
-Placeholder distractor 2
+The function references external variables transparently
 
 ---
 
-Placeholder distractor 3
+The function's name clearly indicates its purpose
 
 #### --answer--
 
-Placeholder answer
+The function can be replaced with its corresponding value without changing the program's behavior
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT a benefit of using pure functions?
 
 #### --distractors--
 
-Placeholder distractor 1
+Easier to test
 
 ---
 
-Placeholder distractor 2
+More predictable behavior
 
 ---
 
-Placeholder distractor 3
+Easier to reason about
 
 #### --answer--
 
-Placeholder answer
+Improved performance in all cases
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the output of this pure function?
+```javascript
+const add = (a, b) => a + b;
+console.log(add(2, 5));
+```
```

Comment:
````suggestion
What is the output of this pure function?

```js
const add = (a, b) => a + b;
console.log(add(2, 5));
```
````
code fences need to be surrounded by empty lines